### PR TITLE
Add url output for broker and channel

### LINKED ISSUE #1093: Add -o url to describe of resources which can be accessed via URL
### Feature request

We've `kn service describe -o url` which returns the service URL. This is to request similar feature for other eventing resources.

- [ ] broker
- [ ] channel

### Use case

User can use `kn broker describe b0 -ourl` etc for the scripting use cases.

### UI Example

```
kn broker describe b0 -ourl
```

```
kn channel describe c0 -ourl
```

/kind good-first-issue

### PR DESCRIPTION
**Pull request by arghya88**
Signed-off-by: Arghya Sadhu <arghya88@gmail.com>

## Changes

*Add -o url to describe of channel and broker resources

## Reference

Fixes #1093

/lint

### CODE REVIEW

**Review by maximilien — APPROVED** [review]
I think this is pretty useful shortcut since one can achieve the same with other output format, e.g., JSON path.

Curious what would happen if the service (or object in question) had multiple URLs, e.g., private and public. I guess you print both...

**Comment by maximilien** [comment]
Seems like unit tests are failing in GH action with:
```bash

        
38
--- FAIL: TestCreateChannelWithTypeFlagInMemoryChannel (0.01s)
39
    mock_helper.go:104: assertion failed: 
40
        --- callArgs[i]
41
        +++ arg
42
          &v1beta1.Channel{
43
          	TypeMeta:   {},
44
          	ObjectMeta: {Name: "pipe"},
45
          	Spec:       {ChannelTemplate: &{TypeMeta: {Kind: "InMemoryChannel", APIVersion: "messaging.knative.dev/v1beta1"}}},
46
          	Status: v1beta1.ChannelStatus{
47
          		ChannelableStatus: v1beta1.ChannelableStatus{
48
          			Status:             {},
49
        - 			AddressStatus:      v1.AddressStatus{Address: &v1.Addressable{URL: s"http://pipe-channel.test"}},
50
        + 			AddressStatus:      v1.AddressStatus{},
51
          			SubscribableStatus: {},
52
          			DeadLetterChannel:  nil,
53
          		},
54
          		Channel: nil,
55
          	},
56
          }
57
        
58
FAIL
59

```

**Review by maximilien — APPROVED** [review]
LGTM
@navidshaikh feel free to merge when ready

File: pkg/kn/commands/broker/describe.go [hunk 1]
```diff
@@ -74,11 +100,15 @@ func describeBroker(out io.Writer, broker *v1beta1.Broker, printDetails bool) er
 	dw := printers.NewPrefixWriter(out)
 	commands.WriteMetadata(dw, &broker.ObjectMeta, printDetails)
 	dw.WriteLine()
-	dw.WriteAttribute("Address", "").WriteAttribute("URL", broker.Status.Address.URL.String())
+	dw.WriteAttribute("Address", "").WriteAttribute("URL", extractURL(broker))
```
Comment (navidshaikh) [inline pkg/kn/commands/broker/describe.go:103]: :+1:

File: pkg/kn/commands/channel/describe.go [hunk 2]
```diff
@@ -28,18 +29,23 @@ import (
 	"knative.dev/client/pkg/printers"
 )
 
+var describeExample = `
+  # Describe a channel 'pipe'
+  kn channel describe pipe
+
+  # Print only channel URL
+  kn channel describe pipe -o url`
+
```
Comment (navidshaikh) [inline pkg/kn/commands/channel/describe.go:38]: :+1:

**Review by navidshaikh — APPROVED** [review]
/lgtm
/approve

Thanks @arghya88 !

**Pull request merged**